### PR TITLE
Remove redundant Python2 numeric relics

### PR DESCRIPTION
**Pull request by eumiro**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [ ] 🧪  Added **tests** for changed code.
- [x] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [ ] 📚  Updated **documentation** for changed code.
- [x] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

Refactor to remove Python2-specific numeric hacks that are no more needed in Python3.

### CODE REVIEW

File: arrow/formatter.py [hunk 1]
```diff
@@ -70,45 +70,45 @@ def _format_token(self, dt, token):
...
         if token == "HH":
             return f"{dt.hour:02d}"
         if token == "H":
-            return str(dt.hour)
+            return f"{dt.hour}"
         if token == "hh":
-            return "{:02d}".format(dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12))
+            return f"{dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12):02d}"
         if token == "h":
             return str(dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12))
```
Comment (jadchaar) [inline arrow/formatter.py:82]: Shouldn't this be changed to an f-string as well?

**Review by jadchaar — CHANGES_REQUESTED** [review]
One small comment.

File: arrow/formatter.py (hunk 1 above)
Comment (eumiro) [reply arrow/formatter.py]: Right, fixed, repushed, thank you. Your `pre-commit` did not complain, so I ignored that line.

Comment (eumiro) [reply arrow/formatter.py]: That expression `dt.hour if 0 < dt.hour < 13 else abs(dt.hour - 12)` could be written as `(dt.hour - 1) % 12 + 1`, but I am not sure whether you'll like it more than the current solution.

Comment (jadchaar) [reply arrow/formatter.py]: I usually prefer modulo to ternary statements. Any preference @systemcatch?

Comment (systemcatch) [reply arrow/formatter.py]: I tend to prefer ternary but honestly don't mind if you guys want to change it.

Comment (jadchaar) [reply arrow/formatter.py]: Thinking about it, I think I actually agree with Chris here. It is more explicit as it is since the hour range is obvious.

Comment (eumiro) [reply arrow/formatter.py]: This is actually a piece of code that never runs on my computer. A day has only 24±1 hours, not 12 :-)

Comment (jadchaar) [reply arrow/formatter.py]: So you are saying one part of the ternary is dead code?

Comment (eumiro) [reply arrow/formatter.py]: No. I want to say that I personally as a user always use 24-hours format, so this piece of code formatting to 12-hours wouldn't be called on my computer. It was just a fun way to close this discussion, and it has nothing to do with this PR.

I totally accept your preference for the ternary and anyway I am happy I was able to contribute to `arrow`.

Comment (jadchaar) [reply arrow/formatter.py]: Thanks for your contributions we really do appreciate them!

**Review by systemcatch — APPROVED** [review]
Ready to go!

**Comment by systemcatch** [comment]
I'll leave merging to @jadchaar in case he has any further comments.

**Review by jadchaar — APPROVED** [review]
LGTM

**Pull request merged**